Python/symtable.c

```diff
@@ -251,6 +256,7 @@ static int symtable_visit_withitem(struct symtable *st, withitem_ty item);
 static int symtable_visit_match_case(struct symtable *st, match_case_ty m);
 static int symtable_visit_pattern(struct symtable *st, pattern_ty s);
 static int symtable_raise_if_annotation_block(struct symtable *st, const char *, expr_ty);
+static int symtable_raise_if_not_coroutine(struct symtable *st, const char *msg, _Py_SourceLocation loc);
```

Comment:
Other functions have a signature that take the statement or expression as aparameter instead of the location directly. So maybe you could do the same here and just call `LOCATION(s)` when you call `SET_ERROR_LOCATION`?

Reply:
That would unnecessarily make this function specific to ``stmt_ty``. There is likely to be an ``expr_ty`` luse case for it as well. 

In general, a function is ideally passed what it actually needs (unless its purpose is to abstract something out).